File: Source/C#/DevnologyFitnesseDojo/DevnologyFitnesseDojo/Fitnesse/Amasun/Book/BuyBookScenario.cs
using System;
using DevnologyFitnesseDojo.Domain.Books;
using DevnologyFitnesseDojo.Domain.Customers;
using DevnologyFitnesseDojo.Domain.Orders;

namespace DevnologyFitnesseDojo.Fitnesse.Amasun.Book
{
    public class BuyBookScenario
    {
        private readonly BookService bookService = new BookService();

        public bool CustomerBuysBooksWithTitle(string customerName, int amount, String title)
        {
            Domain.Books.Book book = Inventory.FindByTitle(title);

            Domain.Customers.Customer customer = CustomerRepository.FindByName(customerName);

            var order = new Order(customer);

            order.AddBook(amount, book);

            bookService.BuyBooks(order);

            return true;
        }
    }
}

File: Source/Java/nl/devnology/fitnesse/amasun/book/BuyBookScenario.java
package nl.devnology.fitnesse.amasun.book;

import nl.devnology.domain.book.Book;
import nl.devnology.domain.book.BookService;
import nl.devnology.domain.book.Inventory;
import nl.devnology.domain.customer.Customer;
import nl.devnology.domain.customer.Customers;

/**
 * @author Erik Pragt
 */
public class BuyBookScenario {
    private BookService bookService = new BookService();


    public boolean customerBuysBooksWithTitle(String customerName, Integer amount, String title) {
        Customer customer = Customers.findByName(customerName);
        Book book = Inventory.get().findByTitle(title);

        bookService.buyBooks(customer, amount, book);

        return true;
    }

    public Integer customerPays(String customerName) {
        return 0;
    }    
}
